renku/api/datasets.py

```diff
@@ -80,7 +79,7 @@ def with_dataset(self, name=None):
                 if path.exists():
                     with path.open('r') as f:
                         source = yaml.load(f) or {}
-                    dataset = Dataset.from_jsonld(source)
+                    dataset = Dataset.from_jsonld(source, __reference__=path)
```

Comment:
This line builds an incomplete model tree where each ```DatasetFile.dataset = None```. Do you have any ideas on how we could easily fix this without additional iteration and setting it manually?

Reply:
we won't need the `dataset` property anymore